python/run-tests.py

```diff
@@ -20,48 +20,90 @@
+    # Run all of the test under test/python directory, each of them
     # has main entry point to execute, which is python's unittest testing
     # framework.
-    test_dir = path.join(root_dir,
-                         path.join("python", path.join("delta", "tests")))
+    python_root_dir = path.join(root_dir, "python")
+    test_dir = path.join(python_root_dir, path.join("delta", "tests"))
     test_files = [os.path.join(test_dir, f) for f in os.listdir(test_dir)
                   if os.path.isfile(os.path.join(test_dir, f)) and
                   f.endswith(".py") and not f.startswith("_")]
+    extra_class_path = path.join(python_root_dir, path.join("delta", "testing"))
+
     for test_file in test_files:
         try:
-            subprocess.check_output(["spark-submit", "--jars", jar_file,
-                                     "--py-files", jar_file, test_file])
+            cmd = ["spark-submit",
+                   "--driver-class-path=%s" % extra_class_path,
+                   "--packages", package, test_file]
+            print("Running tests in %s\n=============" % test_file)
+            print("Command: %s" % str(cmd))
+            run_cmd(cmd, stream_output=True)
         except:
-            err_msg = ""
-            e = sys.exc_info()[1]
-            if isinstance(e, subprocess.CalledProcessError):
-                err_msg = e.output
-            raise Exception("Failed test %s: %s" % (test_file, err_msg))
-
+            print("Failed tests in %s" % (test_file))
+            raise
 
 def prepare(root_dir):
     # Build package with python files in it
     sbt_path = path.join(root_dir, path.join("build", "sbt"))
-    subprocess.call([sbt_path, "clean", "++ 2.11.12 spPackage"])
-    target_dir = path.join(root_dir, "target")
-    jar_files = []
-    for root, dirnames, filenames in os.walk(target_dir):
-        for filename in fnmatch.filter(filenames, '*.jar'):
-            jar_files.append(path.join(root, filename))
-    if len(jar_files) > 1:
-        raise Exception("Multiple JAR files found in %s: %s "
-                        % (target_dir, jar_files.join(", ")))
-    elif len(jar_files) == 0:
-        raise Exception("No JAR file found in %s" % target_dir)
-    return jar_files[0]
+    try:
+        shutil.rmtree(os.path.expanduser("~/.ivy2/cache/io.delta"))
+        print("Deleted ivy2 cache")
+    except:
+        print("ivy2 cache for delta-core is already absent.")
+
+    try:
+        shutil.rmtree(os.path.expanduser("~/.m2/repository/io/delta/"))
+        print("Deleted m2 cache")
+    except:
+        print("m2 cache for delta-core is already absent.")
+    run_cmd([sbt_path, "clean", "++ 2.11.12 publishM2"], stream_output=True)
+
+    # Get current release which is required to be loaded
+    version = '0.0.0'
+    try:
+        with open(os.path.join(root_dir, "version.sbt")) as fd:
+            version = fd.readline().split('"')[1]
+    except:
+        raise Exception("Could not find current release version" +
```

Comment:
You are eating up the original exception ... how would you debug this when something fails?